python/cudf/cudf/core/algorithms.py

```diff
@@ -59,3 +62,59 @@ def factorize(values, sort=False, na_sentinel=-1, size_hint=None):
     values.name = name
 
     return labels, cats.values if return_cupy_array else Index(cats)
+
+
+def linear_interpolation(to_interp):
+    """
+    Interpolate over a float column. Implicitly assumes that values are
+    evenly spaced with respect to the x-axis, for example the data
+    [1.0, NaN, 3.0] will be interpolated assuming the NaN is half way
+    between the two valid values, yielding [1.0, 2.0, 3.0]
+    """
+
+    to_interp._index = RangeIndex(start=0, stop=len(to_interp), step=1)
+    return index_or_values_interpolation(to_interp)
+
+
+def index_or_values_interpolation(to_interp):
+    """
+    Interpolate over a float column. assumes a linear interpolation
+    strategy using the index of the data to denote spacing of the x
+    values. For example the data and index [1.0, NaN, 4.0], [1, 3, 4]
+    would result in [1.0, 3.0, 4.0]
+    """
+    colname = list(to_interp._data.keys())[0]
+    to_interp._data[colname] = (
+        to_interp._data[colname].astype("float64").fillna(np.nan)
+    )
+
+    col = to_interp._data[colname]
+
+    # figure out where the nans are
+    mask = cp.isnan(col)
+
+    # trivial case
+    if mask.all():
+        return col
+
+    mask = as_column(~mask)
+    known_x_and_y = to_interp._apply_boolean_mask(mask)
+
+    known_x = cp.asarray(known_x_and_y._index._column)
+    known_y = cp.asarray(known_x_and_y._data.columns[0])
+
+    result = cp.interp(cp.asarray(to_interp._index), known_x, known_y)
+
+    # find the first nan
+    first_nan_idx = as_column(mask).find_first_value(1)
```

Comment:
anyone got any better ideas for this? :(

Reply:
```suggestion
    first_nan_idx = (mask == 1).argmax().item()
```

Reply:
Note that this suggestions assumes that you apply my previous suggestion where I inline the `mask = as_column(~mask)`. My code here assumes that the `mask` at this point is still a `cupy` array.